src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs

```diff
@@ -1005,18 +1004,19 @@ private SoapArrayInfo ParseArrayType(string value)
             int len = charsLength - pos - 2;
             if (len > 0)
             {
-                string lengthString = new string(chars, pos + 1, len);
+                ReadOnlySpan<char> lengthStringSpan = value.AsSpan(pos + 1, len);
                 try
                 {
-                    soapArrayInfo.length = int.Parse(lengthString, CultureInfo.InvariantCulture);
+                    if (!int.TryParse(lengthStringSpan, CultureInfo.InvariantCulture, out soapArrayInfo.length))
```

Comment:
is try..catch needed now? I don't think OOM is catchable anyway

Reply:
cc: @StephenMolloy @HongGit since this is touching your area

Reply:
Yeah, it doesn't seem like the try/catch is needed now.

Reply:
try/catch removed